Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -388,43 +393,93 @@ public class ProgramBuilder {
     /// Append a splice from another program.
     public func splice(from program: Program, at index: Int) {
         var idx = index
-        
+
+        // As programs are ref counted, we create copy such that we do not
+        // modify the one in the corpus.
+        let workingCopy = program.copy()
+
+        // The placeholder variable is the next free variable in the victim program.
+        var nextFreeVariableCounter = workingCopy.lastVariable + 1
+        func nextFreeVariable() -> Variable {
+           nextFreeVariableCounter += 1
+           return Variable(number: nextFreeVariableCounter - 1)
+        }
+
+        // Get types of the other program
+        var ai = AbstractInterpreter(for: self.fuzzer)
+
+        for instr in workingCopy {
+            ai.execute(instr)
+        }
+
+        beginAdoption(from: workingCopy)
+
         // Determine all necessary input instructions for the choosen instruction
         // We need special handling for blocks:
         //   If the choosen instruction is a block instruction then copy the whole block
         //   If we need an inner output of a block instruction then only copy the block instructions, not the content
         //   Otherwise copy the whole block including its content
         var needs = Set<Int>()
         var requiredInputs = VariableSet()
-        
+
+        // This maps victim instruction indices to victim : host variable remap
+        // Instead of calling adopt and then using nextvar if the variable is
+        // not in the varMaps map, we do the adoption manually.
+        func rewireOrKeepInputs(instruction: Instruction) {
+            var inputs: [Variable] = Array(instruction.inputs)
+            var needInputs: [Variable] = []
+            for inputIdx in 0..<instruction.inputs.count {
```

Comment:
Here you could use [`.enumerated()`](https://developer.apple.com/documentation/swift/array/1687832-enumerated) to get both the value and the index

Reply:
👍